oidcclient/jwk.go

```diff
@@ -51,12 +50,16 @@ func NewOIDCTenant(httpClient *http.Client, targetIss *url.URL) (*OIDCTenant, er
 }
 
 // GetJWKs returns the validation keys either cached or updated ones
-func (ks *OIDCTenant) GetJWKs() (jwk.Set, error) {
-	if time.Now().Before(ks.jwksExpiry) {
-		return ks.jwks, nil
-	}
+func (ks *OIDCTenant) GetJWKs(zoneID string) (jwk.Set, error) {
+	isZoneAccepted, ok := ks.acceptedZoneIds[zoneID]
 
-	updatedKeys, err, _ := ks.singleFlight.Do("updateKeys", ks.updateKeys)
+	if time.Now().Before(ks.jwksExpiry) && ok {
+		if isZoneAccepted {
+			return ks.jwks, nil
+		}
+		return nil, fmt.Errorf("severe security issue: zone_uuid %v is still not accepted", zoneID)
+	}
+	updatedKeys, err := ks.updateKeys(zoneID)
```

Comment:
We should add a `sync.RWMutex` and update the `ks` in the `updateKeys` func.

Reply:
yep, done.